src/Symfony/Component/DependencyInjection/Tests/Compiler/DecoratorServicePassTest.php

```diff
@@ -144,6 +146,21 @@ public function testProcessMovesTagsFromDecoratedDefinitionToDecoratingDefinitio
         $this->assertEquals(array('bar' => array('attr' => 'baz'), 'foobar' => array('attr' => 'bar')), $container->getDefinition('baz')->getTags());
     }
 
+    public function testAutowire()
+    {
+        $container = new ContainerBuilder();
+        $container->register(Bar::class, Bar::class);
+        $container
+            ->register(BarDecorator::class, BarDecorator::class)
+            ->setDecoratedService(Bar::class)
+            ->setAutowired(true)
+        ;
+
+        $this->process($container);
+
+        $this->assertSame('Symfony\Component\DependencyInjection\Tests\Fixtures\BarDecorator.inner', (string) $container->getDefinition(BarDecorator::class)->getArgument(1));
```

Comment:
Mutation testing would hit you hard on this. 
I can see like 8 new conditions in src code, yet only one path is executed in tests.
You are testing the positive scenario, which is really great.
Yet, please add negative test scenario as well, to be sure that it wont autowire if any of requirements for autowiring is not fulfilled.

Reply:
Yes I will :)